server-spi-private/src/main/java/org/keycloak/userprofile/DefaultAttributes.java

```diff
@@ -205,6 +209,35 @@ public boolean validate(String name, Consumer<ValidationError>... listeners) {
         return result == null;
     }
 
+    /**
+     * In case there are unmanaged attributes or attributes that don't have a length restrictions,
+     * add a default length restriction to avoid a denial of service by a caller.
+     */
+    private static void limitLengthOnAttributesWithNoLengthRestriction(String name, List<AttributeMetadata> metadatas) {
+        boolean lengthValidatorExists = false;
+
+        for (AttributeMetadata metadata : metadatas) {
+            for (AttributeValidatorMetadata validator : metadata.getValidators()) {
+                if (validator.getValidatorId().equals(LengthValidator.ID)) {
+                    lengthValidatorExists = true;
+                    break;
```

Comment:
Can't this whole thing be a simplified a bit by just returning from here instead of breaking out of two loops just to skip the rest of the method? i.e. remove the flag, and return if the id matches here.

Reply:
Yes, that work work. I'll add anther commit. The data is in-memory IMHO already, anyway.

Reply:
Commit added. Build will hopefully be green. Ready for a final round of review?